doc/automated-init.md

```diff
@@ -0,0 +1,113 @@
+# Provisioning Resin.io devices in automated (non-interactive mode)
+
+This document describes how to run the `device init` command in non-interactive mode.
+
+It requires collecting some preliminary information _once_.
+
+The final command to provision the device looks like this:
+
+```bash
+resin device init --app APP_ID --os-version OS_VERSION --drive DRIVE --config "$(cat CONFIG_FILE)" --yes
```

Comment:
It's definitely not the normal bash convention though. I can think of lots of tools that take config filenames on the command line, but not a single one that takes the full config raw.

I'm not convinced the flexibility is going to be frequently necessary for this, but I can see how it'd be nice. The normal convention for that is to pass `-` though, and interpret that as 'read this from stdin'.

Alternatively, even if you don't want to explicitly support that pattern, there's a whole bunch of standard bash conventions for creating temporary names that refer to pipes, which let you support exactly this case without any effort in our code at all: https://serverfault.com/questions/398514/pass-a-pipe-to-a-command-that-expects-a-filename